hollow/src/main/java/com/netflix/hollow/core/memory/encoding/HashCodes.java

```diff
@@ -168,13 +168,40 @@ public static int hashInt(int key) {
         return key;
     }
 
-    public static int hashTableSize(int numElements) {
+    /**
+     * Determine size of hash table capable of storing the specified number of elements with a load
+     * factor applied.
+     *
+     * @param numElements number of elements to be stored in the table
+     * @return size of hash table, always a power of 2
+     * @throws IllegalArgumentException when numElements is negative
+     */
+    public static int hashTableSize(int numElements) throws IllegalArgumentException {
+        if (numElements < 0) {
+            throw new IllegalArgumentException("cannot be negative; numElements="+numElements);
+        } else if (numElements > HASH_TABLE_MAX_SIZE) {
```

Comment:
I *think* it's safe to always treat this as an error. For example, `HollowPrimaryKeyIndex` uses this to construct a `FixedLengthElementArray` for an open-addressing hash table and we cannot store more elements in it than the maximum hash table size.